hip/multigrid/amgx_pgm_kernels.hip.cpp

```diff
@@ -62,25 +68,62 @@ namespace hip {
 namespace amgx_pgm {
 
 
+constexpr int default_block_size = 512;
+
+
+#include "common/multigrid/amgx_pgm_kernels.hpp.inc"
+
+
 template <typename IndexType>
 void match_edge(std::shared_ptr<const HipExecutor> exec,
                 const Array<IndexType> &strongest_neighbor,
-                Array<IndexType> &agg) GKO_NOT_IMPLEMENTED;
+                Array<IndexType> &agg)
+{
+    const auto num = agg.get_num_elems();
+    const dim3 grid(ceildiv(num, default_block_size));
+    hipLaunchKernelGGL(kernel::match_edge_kernel, dim3(grid),
+                       dim3(default_block_size), 0, 0, num,
+                       strongest_neighbor.get_const_data(), agg.get_data());
+}
 
 GKO_INSTANTIATE_FOR_EACH_INDEX_TYPE(GKO_DECLARE_AMGX_PGM_MATCH_EDGE_KERNEL);
 
 
 template <typename IndexType>
 void count_unagg(std::shared_ptr<const HipExecutor> exec,
-                 const Array<IndexType> &agg,
-                 IndexType *num_unagg) GKO_NOT_IMPLEMENTED;
+                 const Array<IndexType> &agg, IndexType *num_unagg)
+{
+    Array<IndexType> active_agg(exec, agg.get_num_elems());
+    const dim3 grid(ceildiv(active_agg.get_num_elems(), default_block_size));
+    hipLaunchKernelGGL(kernel::activate_kernel, dim3(grid),
+                       dim3(default_block_size), 0, 0,
+                       active_agg.get_num_elems(), agg.get_const_data(),
+                       active_agg.get_data());
+    *num_unagg = reduce_add_array(exec, active_agg.get_num_elems(),
+                                  active_agg.get_const_data());
+}
 
 GKO_INSTANTIATE_FOR_EACH_INDEX_TYPE(GKO_DECLARE_AMGX_PGM_COUNT_UNAGG_KERNEL);
 
 
 template <typename IndexType>
 void renumber(std::shared_ptr<const HipExecutor> exec, Array<IndexType> &agg,
-              IndexType *num_agg) GKO_NOT_IMPLEMENTED;
+              IndexType *num_agg)
+{
+    const auto num = agg.get_num_elems();
+    Array<IndexType> agg_map(exec, num + 1);
+    components::fill_array(exec, agg_map.get_data(), agg_map.get_num_elems(),
+                           zero<IndexType>());
```

Comment:
nit: can be simplified.
```suggestion
    agg_map.fill(zero<IndexType>());
```